File: inventory-management/src/main/java/com/inventory_management/service/ProductService.java
package com.inventory_management.service;

import com.inventory_management.dto.request.ProductRequestDTO;
import com.inventory_management.dto.response.ProductResponseDTO;
import org.springframework.data.domain.Page;

public interface ProductService {

    ProductResponseDTO createProduct(ProductRequestDTO request);

    Page<ProductResponseDTO> getAllProducts(
        int page,
        int size,
        String sortBy,
        String sortDir
);

Page<ProductResponseDTO> searchProducts(

        String keyword,

        int page,

        int size,

        String sortBy,

        String sortDir

);

Page<ProductResponseDTO> getActiveProducts(
        int page,
        int size,
        String sortBy,
        String sortDir
);

Page<ProductResponseDTO> getInactiveProducts(
        int page,
        int size,
        String sortBy,
        String sortDir
);

    void restoreProduct(Integer id);

    ProductResponseDTO getProductById(Integer productId);

    ProductResponseDTO updateProduct(Integer productId, ProductRequestDTO request);

    void deleteProduct(Integer productId);

}

File: inventory-management-frontend/src/services/productService.js
import axiosInstance from "../api/axiosConfig";


const productService = {


    async getProducts(params){

        const response = await axiosInstance.get(
            "/products",
            {
                params
            }
        );

        return response.data;

    },



    async searchProducts(params){

        const response = await axiosInstance.get(
            "/products/search",
            {
                params
            }
        );

        return response.data;

    },



    async createProduct(data){

        const response = await axiosInstance.post(
            "/products",
            data
        );

        return response.data;

    },



    async updateProduct(id,data){

        const response = await axiosInstance.put(

            `/products/${id}`,

            data

        );

        return response.data;

    },



    async deleteProduct(id){

        await axiosInstance.delete(

            `/products/${id}`

        );

    },

    async getActiveProducts(params){

        const response = await axiosInstance.get(
            "/products/active",
            {params}
        );

        return response.data;

    },

    async getInactiveProducts(params){

    const response = await axiosInstance.get(
        "/products/inactive",
        {params}
    );

    return response.data;

    },

    async restoreProduct(id){

    await axiosInstance.patch(
        `/products/${id}/restore`
    );

}

};


export default productService;